src/main/java/org/openstreetmap/atlas/geography/Polygon.java

```diff
@@ -248,6 +253,73 @@ public Location interiorCenter()
         return new JtsLocationConverter().backwardConvert(point.getCoordinate());
     }
 
+    /**
+     * @param nCount
+     *            Number of sides
+     * @param threshold
+     *            {@link Angle} threshold that decides whether a {@link Heading} difference between
+     *            segments should be counted towards heading change count or not
+     * @return true if this {@link Polygon} has approximately n sides while ignoring {@link Heading}
+     *         differences between inner segments that are below given threshold.
+     */
+    public boolean isApproximatelyNSided(final int nCount, final Angle threshold)
```

Comment:
How about changing `nCount` to `expectedNumberOfSides` or something similar to make it more explicit? The same way you have `expectedHeadingChangeCount` below, which is really easy to read.